client/src/main/java/io/pravega/keycloak/client/KeycloakAuthzClient.java

```diff
@@ -104,6 +115,51 @@ public String getRPT() {
         return token.getToken();
     }
 
+    /**
+     * Predicate to determine what is retryable and what is not.
+     * HttpResponseException with error code of 400, 401, 403 are not retryable.
+     * All other HttpResponseException are retryable as well as java.net.ConnectException.
+     * All others are not retryable.
+     * @return
+     */
+    private static Predicate<Throwable> isRetryable() {
+        return e -> {
+            Throwable rootCause = unwrap(e);
+            if (rootCause instanceof HttpResponseException) {
+                int statusCode = ((HttpResponseException) e).getStatusCode();
+                if (statusCode == HttpStatus.SC_BAD_REQUEST ||
+                        statusCode == HttpStatus.SC_UNAUTHORIZED ||
+                        statusCode == HttpStatus.SC_FORBIDDEN ) {
+                    // these are authN or authZ related errors.
+                    LOG.error("Non retryable HttpResponseException: {}", statusCode);
```

Comment:
Should log the exception trace too, otherwise, it might become hard to debug issues.